e2e_tests/tests/conftest.py

```diff
@@ -154,3 +155,16 @@ def test_start_timer(request: SubRequest) -> Iterator[None]:
         # This ends up concatenated to the line pytest prints containing the test file and name.
         print("starting at", time.strftime("%Y-%m-%d %H:%M:%S"))
     yield
+
+
+@pytest.fixture
+def profile_test(record_property: Callable[[str, object], None]) -> Callable[[int], None]:
+    """
+    Returns a method that allows profiling of test run for certain system metrics
+    and records to JUnit report.
+
+    Currently retrieves metrics by trial (assumes one trial per experiment) using
+    profiler API.
+    """
+
+    return ProfileTest(record_property=record_property).record
```

Comment:
I think it makes more sense to use just a higher order function that takes record_property and returns ProfileTest.record instead of returning a method on a class.

Reply:
hmm... so i want the metrics to be configurable from the conftest config, like: 
`ProfileTest(record_property=record_property, metrics={SysMetricName.DISK_IOPS_METRIC: ["avg", "max"]}).record`
but also keep the fixture that's returned really simple in signature, i.e tests should only need to pass in trial_id in order to use it. is there a cleaner way to do this with higher order functions?

Reply:
I have I don't seen anywhere it is being used like that but if you wanted it configurable you could just have a signature like
```
def recorder_maker(record_property, metrics):
    def record(self, trial_id):
        ...
     return record
```
i mean, i'm not sure there is much of a difference in capturing the state as an instance var in a class vs a closure like there, other than the original style nit that it feels weird to me to return a class instance method without the class.

Reply:
makes sense, i like this better too. changed.